lib/ansible/modules/source_control/gitlab_project.py

```diff
@@ -151,141 +165,132 @@
-            if owner == project_owner:
-                return result['id']
-
-    def getUserId(self, user_name):
-        user_data = self._gitlab.getusers(search=user_name)
-
-        for data in user_data:
-            if 'id' in data:
-                return data['id']
-        return self._gitlab.currentuser()['id']
-
-    def to_bool(self, value):
-        if value:
-            return 1
-        else:
-            return 0
+            return False
 
-    def updateProject(self, group_name, arguments):
-        project_changed = False
-        project_name = arguments['name']
-        project_id = self.getProjectId(group_name, project_name)
-        project_data = self._gitlab.getproject(project_id=project_id)
+    '''
+    @param namespace Namespace Object (User or Group)
+    @param arguments Attributs of the project
+    '''
+    def createProject(self, namespace, arguments):
+        if self._module.check_mode:
+                self._module.exit_json(changed=True, result="Project should have been created.")
+
+        arguments['namespace_id'] = namespace.id
+        try:
+            project = self._gitlab.projects.create(arguments)
+        except (gitlab.exceptions.GitlabCreateError) as e:
+            self._module.fail_json(msg="Failed to create a project: %s " % to_native(e))
+
+        return project
+
+    '''
+    @param project Project Object
+    @param arguments Attributs of the project
+    '''
+    def updateProject(self, project, arguments):
+        changed = False
 
         for arg_key, arg_value in arguments.items():
-            project_data_value = project_data[arg_key]
-
-            if isinstance(project_data_value, bool) or project_data_value is None:
-                to_bool = self.to_bool(project_data_value)
-                if to_bool != arg_value:
-                    project_changed = True
-                    continue
-            else:
-                if project_data_value != arg_value:
-                    project_changed = True
-
-        if project_changed:
-            if self._module.check_mode:
-                self._module.exit_json(changed=True)
-            return self._gitlab.editproject(project_id=project_id, **arguments)
-        else:
-            return False
+            if arguments[arg_key] is not None:
+                if getattr(project, arg_key) != arguments[arg_key]:
+                    setattr(project, arg_key, arguments[arg_key])
+                    changed = True
+
+        return (changed, project)
+
+    def deleteProject(self):
+        if self._module.check_mode:
+            self._module.exit_json(changed=True, result="Project should have been deleted.")
+
+        project = self.projectObject
+
+        return project.delete()
+
+    '''
+    @param name Name of the groupe
+    @param full_path Complete path of the Group including parent group path. <parent_path>/<group_path>
+    '''
+    def findGroup(self, name, full_path):
```

Comment:
`findGroup` and `findProject` are repeated in a few places, worth moving out into `module_utils`? There's a `gitlab` library out there already that won't be needed anymore, you could use that for generic stuff.

Reply:
I will create a function in gitlab library